src/base/static/components/organisms/main-map.js

```diff
@@ -0,0 +1,377 @@
+import { createGeoJSONFromCollection } from "../../utils/collection-utils";
+
+import { mapConfigSelector } from "../../state/ducks/map-config";
+import {
+  mapLayersBasemapSelector,
+  mapSizeValiditySelector,
+  setMapSizeValidity,
+  setMapPosition,
+} from "../../state/ducks/map";
+
+import constants from "../../constants";
+
+const Util = require("../../js/utils.js");
+
+class MainMap extends Component {
+  constructor(props) {
+    super(props);
+    this._router = props.router;
+    this._places = props.places;
+    this._layers = props.mapConfig.layers;
+
+    let MapProvider;
+    switch (props.mapConfig.provider) {
+      // Add other provider types here as they become available
+      case "mapboxgl":
+        MapProvider = MapboxGLProvider;
+        break;
+      default:
+        MapProvider = MapboxGLProvider;
+        break;
+    }
+
+    // Bind visiblity event for custom layers
+    $(Shareabouts).on("visibility", async (evt, layer) => {
+      await this.createLayerFromConfig(layer);
+    });
+
+    this._map = MapProvider(props.container, props.mapConfig.options);
+
+    // TEMPORARY: Manually trigger the visibility of layers for testing
+    this._map.on("load", () => {
+      $(Shareabouts).trigger("visibility", [this._layers[4], true, true]);
+    });
+
+    if (props.mapConfig.geolocation_enabled) {
+      // TODO
+    }
+  }
+
+  componentDidUpdate() {
+    if (!this.props.isValidSize) {
+      this._map.invalidateSize();
+      this.props.setMapSizeValidity(true);
+    }
+  }
+
+  componentDidMount() {
+    this._map.on("dragend", () => {
+      Util.log(
+        "USER",
+        "map",
+        "drag",
+        this._map.getBBoxString(),
+        this._map.getZoom(),
+      );
+      this.props.onDragend();
+    });
+    this._map.on("zoomend", () => {
+      Util.log("APP", "zoom", this._map.getZoom());
+      this.props.onZoomend();
+    });
+    this._map.on("movestart", this.props.onMovestart);
+    this._map.on("moveend", () => {
+      Util.log("APP", "center-lat", this._map.getCenter().lat);
+      Util.log("APP", "center-lng", this._map.getCenter().lng);
+      this.props.setMapPosition({
+        center: this._map.getCenter(),
+        zoom: this._map.getZoom(),
+      });
+      this.props.onMoveend();
+    });
+
+    emitter.addListener("place-collection:loaded", collectionId => {
+      this.addPlaceCollectionLayer(collectionId);
+    });
+    emitter.addListener("place-collection:add-place", collectionId => {
+      this.updatePlaceCollectionLayer(collectionId);
+    });
+    emitter.addListener(
+      "place-collection:focus-place",
+      ({ collectionId, modelId }) => {
+        this.updatePlaceCollectionLayer(collectionId, modelId);
+      },
+    );
+    emitter.addListener("place-collection:unfocus-all-places", () => {
+      Object.keys(this._places).forEach(collectionId => {
+        this.updatePlaceCollectionLayer(collectionId);
+      });
+    });
+
+    emitter.addListener(
```

Comment:
Here we add a few listeners for the different map transition events that might be fired in the `AppView` by routing to a story chapter.